### PR TITLE
Signal Handling in Cpp

### LINKED ISSUE #3684: Signal Handling in C++
### Proposed title of article
Signal Handling in C++

### Introduction paragraph (2-3 paragraphs):
Signals are the interrupts delivered to a process by the operating system which can terminate a program prematurely. You can generate interrupts by pressing Ctrl+C on a UNIX, LINUX, Mac OS X or Windows system. C++ signal-handling library provides function `signal` to trap unexpected events. In this tutorial, we will learn how to handle signals in C++ using the `signal()` and the `raise()` function.

### Key takeaways: 
By the end of this article the reader should have learned the following:

- Signals that are defined in C++ header file
- The signal() Function
- The raise() Function

### References: 
N/A

### PR DESCRIPTION
**Pull request by fely545**
### Article checklist

> If you have not already, please go over our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for more tips and tricks on how to improve your overall technical writing so reviews are swifter, increase the chance of a payout and provide more value to our readers.

> For first-time contributors and for more details on our submission guidelines, **see our [Contributing Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**.

#### Formatting and structure checks
- [X] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
- [X] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [X] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [X] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**

- [X] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [X] Is the article you are submitting an in-depth and unique article? Does it go beyond what is in the official docs and what is covered in other blog sites. See these articles as [examples](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md).
- [X] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [X] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**

- [X] Have you included a hero image that is 600x400 pixels and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
      
#### Grammar & plagiarism checks
- [X] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) - [Grammarly](https://grammarly.com) 
- [X] Please place your article through a 3rd party plagiarism checker. We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [X] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [X] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [X] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**
- [X] Have you checked your code runs correctly and you've highlighted all necessary dependencies for installation?
- [X] Are the software programs and packages you're highlighting in your article up to date, using current versions and not deprecated?

`This closes #3684

> Finally, delete the article checklist notes in blockquotes and submit your PR. We look forward to reviewing your article.

### CODE REVIEW

**Comment by ahmadmardeni1** [comment]
Good afternoon and thank you for submitting your article for publishing. 
With the recent increase in article volume out team has been working diligently to review every article. Your article has been entered into our queue and should be reviewed (and published pending any changes) as soon as a peer reviewer is finished with the ones in the queue before it. 🚀
Thank you.

**Comment by Kayere** [comment]
Hello @fely545,

 Upon running your article through our 3rd party plagiarism checker it seemed to raise a few flags and the % was higher than we typically accept.

Please see attached PDF - and revisit the article to ensure we are contributing wholly unique and original content.

Be sure to see our resources page to see more info on plagiarism and what is considered as such.

[Signal Handling in Cpp #5364 Copyleaks Report.pdf](https://github.com/section-engineering-education/engineering-education/files/7692434/Signal.Handling.in.Cpp.5364.Copyleaks.Report.pdf)

File: content/articles/signal-handling-in-c++/index.md [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+### Signal handling in C++
+We understand that signals are interrupts that stop an OS in its task to attend to another task, and to stop these actions, the signals need to be handled.
+
+In C++, signals are provided by the language. It catch the interrupting signals and solve the problem that causes the program to pause while executing its task.
+
+Below is the C++ signal library, signals that a developer can work with and their operations.
+
+#### Signal operations
+1. `SIGINT`(Signal Interrupt) - This signal is used to generate an interactive attention signal as a "program interrupt" at the receiver end. When the OS generates an interrupt, this C++ signal interrupts the interrupt and allows the ongoing task to continue.
+<!--  -->
```
Comment (geoffrey45) [inline content/articles/signal-handling-in-c++/index.md:40]: This statement: ` ... this C++ signal interrupts the interrupt ...` is a bit confusing. Please rephrase it to remove the confusion between the word "interrupt".

File: content/articles/signal-handling-in-c++/index.md [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+topics: [Security]
+excerpt_separator: <!--more-->
+images:
+  - url: /engineering-education/signal-handling-in-cpp/hero.jpg
+    alt: Signal Handling In C++
+---
+There are interruptions that come up in your program, which are delivered to a process that runs on an operating system and pauses the running process. These interrupts are called signals.
+<!--more-->
+The signals can be generated by pressing `Ctrl+C` on various operating systems (Windows, Mac OS X, LINUX, or UNIX).
+
```
Comment (geoffrey45) [inline content/articles/signal-handling-in-c++/index.md:19]: The keys combination `CTRL + C` sends an interrupt signal on most cases. Please mention this.

**Review by geoffrey45 — CHANGES_REQUESTED** [review]
@fely545 Please check the comments below.

**Comment by fely545** [comment]
@geoffrey45 Done with the changes.

**Review by geoffrey45 — APPROVED** [review]

**Comment by hectorkambow** [comment]
jumping on this one
